uaclient/config.py

```diff
@@ -477,10 +484,15 @@ def delete_cache_key(self, key: str) -> None:
         cache_path = self.data_path(key)
         self._perform_delete(cache_path)
 
-    def delete_cache(self):
-        """Remove configuration cached response files class attributes."""
+    def delete_cache(self, delete_permanent: bool = False):
+        """
+        Remove configuration cached response files class attributes.
+
+        :param delete_permanent: even delete the "permanent" files
+        """
         for path_key in self.data_paths.keys():
-            self.delete_cache_key(path_key)
+            if delete_permanent or not self.data_path(path_key).permanent:
```

Comment:
The `self.data_path` call will return a string here, so we cannot call the `permanent` attribute like this

Reply:
Oof good catch!